OpenDreamRuntime/Procs/Native/DreamProcNativeIcon.cs

```diff
@@ -17,6 +18,41 @@ static class DreamProcNativeIcon {
             return new DreamValue(dreamIconObject.Description.Height);
         }
 
+        [DreamProc("Insert")]
+        [DreamProcParameter("new_icon", Type = DreamValue.DreamValueType.DreamObject)]
+        [DreamProcParameter("icon_state", Type = DreamValue.DreamValueType.String)]
+        [DreamProcParameter("dir", Type = DreamValue.DreamValueType.Float)]
+        [DreamProcParameter("frame", Type = DreamValue.DreamValueType.Float)]
+        [DreamProcParameter("moving", Type = DreamValue.DreamValueType.Float)]
+        [DreamProcParameter("delay", Type = DreamValue.DreamValueType.Float)]
+        public static DreamValue NativeProc_Insert(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
+            //TODO Figure out what happens when you pass the wrong types as args
 
+            arguments.GetArgument(0, "new_icon").TryGetValueAsDreamObject(out var new_icon);
```

Comment:
The ref says this could also be an icon resource. This should either handle that, or throw an unimplemented exception.